quota/quota.go

```diff
@@ -0,0 +1,180 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package quota
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+
+	"github.com/snapcore/snapd/gadget/quantity"
+)
+
+// Group is a quota group of snaps, services or sub-groups that are all subject
+// to specific resource quotas. Current quota resource types supported are just
+// memory, but this can be expanded in the future.
+type Group struct {
+	// Name is the name of the quota group. Certain names are reserved and have
+	// special meaning such as user.* or system.* or snapd.*, but are otherwise
```

Comment:
we need to introduce a naming.ValidateQuotaGroup to define exactly what names we allow

Reply:
Added